### ISSUE TITLE
Auto-filling the master password form using Apple Keychain writes the user name into key file field

### ISSUE THREAD

**Issue by keepassium** [keepassium/KeePassium#41]
Steps to replicate (approximate):
- Create an entry for KeePassium in Apple Keychain
- Enable Keychain in system's AutoFill settings
- Open KeePassium, choose a database
- Use AutoFill/Keychain to fill the master key form

Expected result: only the password field is filled.

Observed result: the key file field is filled with the user name from the Keychain.

[[Thanks, /u/Silunare](https://www.reddit.com/r/KeePassium/comments/ckiezu/some_minor_issues/)]

**Comment by Silun** [comment]
In essence yes, but I should try to be more clear about it:

The keychain does not allow the manual creation of an app entry, but it does allow manual creation of website entries. So that is what I did: 

I added a website KeePassium.com with the password and a 1-letter username.

Then in the password field in KeePassium, I add the details via the keychain and it would add the username in the keyfile field, promoting me to choose the correct keyfile again.

Now what I did there was a workaround for the actual issue at hand: KeePassium doesn't register with keychain.

For example, when I first open a new app and enter it's login details, keychain might ask me whether I want to save those credentials into the keychain. This creates an app entry in the keychain that can not be done manually AFAIK. It is this kind of registration that is the underlying issue. 


It also leads to another issue, which I'm going to present here as well because the fix for both is most likely one and the same:
When I open eg PayPal and choose KeePassium to get my credentials, I am already presented only with PayPal, so probably PayPal's app passes it's name to KeePassium. 

KeePassium itself however doesn't do this, when I choose KeePassium to find it's own database password, it just shows me a list of all entries, rather than just those with KeePassium as part of the name.
